genie/__main__.py

```diff
@@ -1,37 +1,58 @@
 #!/usr/bin/env python
-import argparse
 import genie
-
-
-def perform_main(syn, args):
-    if 'func' in args:
-        try:
-            args.func(syn, args)
-        except Exception:
-            raise
+import synapseclient
+import logging
+logger = logging.getLogger('genie')
+
+
+def synapse_login(username=None, password=None):
+    """
+    This function logs into synapse for you if credentials are saved.
+    If not saved, then user is prompted username and password.
+
+    :returns:     Synapseclient object
+    """
+    try:
+        syn = synapseclient.login(silent=True)
+    except Exception:
+        if username is None and password is None:
+            raise ValueError(
+                "Please specify --syn_user, --syn_pass the first time "
```

Comment:
Add link to our docs about login options - use keyring, or config, or a 'rememberMe'